### PR TITLE
Add python 3.9 to CI and setup.py

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
         'spark': ['pyspark>=2.4.0'],
         'mlflow': ['mlflow>=1.0'],
     },
-    python_requires='>=3.5,<3.9',
+    python_requires='>=3.5,<4.0',
```
Comment (HyukjinKwon) [inline setup.py:49]: I would do it as `3.10` instead of `4.0`.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -118,6 +118,11 @@ jobs:
             pandas-version: 1.1.3
             pyarrow-version: 1.0.1
             default-index-type: 'distributed-sequence'
+          - python-version: 3.9
```
Comment (HyukjinKwon) [inline .github/workflows/master.yml:121]: I think we can just change the Python version in the existing matrix above instead of adding a new matrix.

Comment (gsheni) [reply .github/workflows/master.yml]: Could you provide an example of this?

File: setup.py (hunk 1 above)
Comment (gsheni) [reply setup.py]: Fixed

File: .github/workflows/master.yml (hunk 2 above)
Comment (HyukjinKwon) [reply .github/workflows/master.yml:121]: For example,

```
          - python-version: 3.6
            spark-version: 2.4.7
            pandas-version: 0.24.2
            pyarrow-version: 0.14.1
            logger: databricks.koalas.usage_logging.usage_logger
          - python-version: 3.6
            spark-version: 2.4.7
            pandas-version: 0.25.3
            pyarrow-version: 0.15.1
            default-index-type: 'distributed-sequence'
          - python-version: 3.7
            spark-version: 2.4.7
            pandas-version: 0.25.3
            pyarrow-version: 0.14.1
          - python-version: 3.7
            spark-version: 2.4.7
            pandas-version: 1.0.5
            pyarrow-version: 0.15.1
            default-index-type: 'distributed-sequence'
          - python-version: 3.8
            spark-version: 3.0.1
            pandas-version: 0.25.3
            pyarrow-version: 0.15.1
          - python-version: 3.9
            spark-version: 3.0.1
            pandas-version: 1.1.3
            pyarrow-version: 1.0.1
            default-index-type: 'distributed-sequence'
```

**Comment by HyukjinKwon** [comment]
@gsheni looks like Python 3.9 is up in Conda. Can you try to fix the build?

**Comment by HyukjinKwon** [comment]
@gsheni can you sync the latest master branch to see if the builds pass now?

**Comment by HyukjinKwon** [comment]
Seems like it fails to install during dependency resolution. Can you resolve this? https://github.com/databricks/koalas/pull/1831/checks?check_run_id=1646674809

**Comment by gsheni** [comment]
Closing this as the commit history is messed up. Will open up a new MR.

**Pull request closed without merging**